src/diamond/handler/cloudwatch.py

```diff
@@ -170,6 +181,11 @@ def process(self, metric):
         metricname = str(metric.getMetricPath())
         timestamp = datetime.datetime.fromtimestamp(metric.timestamp)
 
+        if self.autoscaling:
+            instance = self.ec2.get_only_instances([self.instance_id])
+            self.log.debug(instance[0].tags['aws:autoscaling:groupName'])
+            autoscaling_group = instance[0].tags['aws:autoscaling:groupName']
```

Comment:
Should this have a guard against the tag possibly missing, to avoid breaking the handler invocation with `KeyError`s?